### PR TITLE
feat: Whats On Page

### LINKED ISSUE #9: Create homepage what's on
- Create production showcase component with placeholder content
   - Ensure that show poster used (A4 dimensions / ratio)
image

- With above component, create a mock what's on section with multiple components
image

### LINKED ISSUE #34: All Upcoming Productions Page
Create all upcoming productions page (and update the link on the homepage to link to it)

Thinking a paginated grid layout (3x*)

### PR DESCRIPTION
**Pull request by atoff**
TODO:

- [x] Refactor out infinite loading as a component

Closes #34

### CODE REVIEW

File: src/components/production/ProductionTile.vue [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    },
+  },
+  methods: {
+    displayStartEnd,
+  },
+};
+</script>
+
+<style scoped lang="scss">
+.production-tile:hover {
```
Comment (streth11) [inline src/components/production/ProductionTile.vue:39]: can this not be tailwind?

File: src/fakeApi/index.js [hunk 2]
```diff
@@ -259,6 +259,8 @@ export function makeServer({ environment = 'development' } = {}) {
         name: 'A Default Production',
       });
 
+      server.createList('ProductionNode', 20);
```
Comment (streth11) [inline src/fakeApi/index.js:262]: fakeAPI goes brrrrrrrrrr

File: src/components/production/ProductionTile.vue (hunk 1 above)
Comment (atoff) [reply src/components/production/ProductionTile.vue:39]: Unfortunately not - we want the test to change colour when the user hovers over any part of the tile. We could do this with tailwind if the show name heading didn't have a manual colour assigned, but this takes priority over the hover colour so you have to do it with css.

Comment (streth11) [reply src/components/production/ProductionTile.vue]: i seeeee

File: src/views/production/UpcomingProductions.vue [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <div class="min-h-full text-white bg-sta-gray">
+    <div class="container">
+      <h1 class="py-4 text-h1">Whats On</h1>
+      <div class="grid grid-cols-1 gap-4 mb-4 sm:grid-cols-2 lg:grid-cols-3">
+        <production-tile
+          v-for="(production, i) in productions"
+          :key="i"
```
Comment (streth11) [inline src/views/production/UpcomingProductions.vue:8]: i -> index   -- we have used index everywhere else in the project

File: src/views/production/UpcomingProductions.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ProductionTile from '@/components/production/ProductionTile.vue';
+export default {
+  name: 'Production',
```
Comment (streth11) [inline src/views/production/UpcomingProductions.vue:36]: upcomingProductions

**Review by streth11 — DISMISSED** [review]
love it

**Review by streth11 — APPROVED** [review]

**Pull request merged**